### PR TITLE
Log error results reported to CompactionCB.

### PR DESCRIPTION
**Pull request by kimoonkim**
Part of https://trello.com/c/ZioYyW9x/354-looks-into-the-tsdb-deferredgroupexception.

I am about to test this with the ec2 integration test as described in http://tiny.pepperdata.com/dashboard-ec2-backend. And possibly add unit test coverage.

### CODE REVIEW

**Comment by kimoonkim** [comment]
There is a unit test coverage for this code path:

https://github.com/pepperdata/opentsdb/blob/next_pepperdata/test/core/TestTSDB.java#L802

```
 public void addPointList_completionCallbackError() throws Exception {
    setupAddPointStorage();

    // Tests that addPointList calls the completion callback with an Exception
    // when there is an error while inserting rows.
    Answer<Deferred<Object>> answer = new Answer<Deferred<Object>>() {

      final List<Deferred<Object>> deferred_results = ImmutableList.of(
          Deferred.fromResult(null),
          Deferred.fromError(new Exception("unittest")),
          Deferred.fromResult(null));
      final Iterator<Deferred<Object>> result_iter = deferred_results.iterator();
      @Override
      public Deferred<Object> answer(InvocationOnMock invocation)
          throws Throwable {
        return result_iter.next();
      }
    };
```

We now  properly log ` Deferred.fromError(new Exception("unittest"))` above:
```
10:45:09.607 [main] WARN  net.opentsdb.core.TSDB - CompactionCB got an error result, adding to 1 errors so far:
java.lang.Exception: unittest
        at net.opentsdb.core.TestTSDB$1.<init>(TestTSDB.java:810) ~[test-classes/:na]
        at net.opentsdb.core.TestTSDB.addPointList_completionCallbackError(TestTSDB.java:807) ~[test-classes/:na]
...
```

Previously, it was just silently ignored except the contribution to the error count.

**Comment by kimoonkim** [comment]
@bpitman and @pepper-ankan, added you guys as reviewers in case you guys want to quickly check the sanity of this. Otherwise, I'm going to merge this in after ec2 integration test is done.

File: src/core/TSDB.java [hunk 1]
```diff
@@ -87,15 +87,22 @@
...
         ++errors;
+        e = (Exception)result;
+        LOG.warn("CompactionCB got an error result, adding to " + errors + " errors so far:",
+            (Exception)e);
       }
       --count;
       if (count <= 0) {
         if (errors > 0) {
-          deferred_completion.callback(new Exception("errors = " + errors));
+          String message = "Reporting a CompactionCB has total " + errors + " errors." +
```
Comment (pepper-ankan) [inline src/core/TSDB.java]: Maybe s/Reporting a CompactionCB has total/Reporting that CompactionCB has a total of/ ?

**Comment by kimoonkim** [comment]
Done with ec2 integration test. Merging in.

**Pull request merged**